Subscriber/ApplePayDirectSubscriber.php

```diff
@@ -115,6 +194,55 @@ public function onFrontendCheckoutPostDispatch(Enlight_Event_EventArgs $args)
         $view->assign('sPayments', $sPayments);
     }
 
+    /**
+     * @param int $customerId
+     * @param int $paymentId
+     * @throws \Doctrine\ORM\ORMException
+     * @throws \Doctrine\ORM\OptimisticLockException
+     */
+    private function updateCustomerDefaultPaymentMethod(int $customerId, int $paymentId) {
+        $repository = $this->modelManager->getRepository(Customer::class);
+        $customer = $repository->find($customerId);
+
+        $customer->setPaymentId($paymentId);
+
+        $this->modelManager->persist($customer);
+        $this->modelManager->flush();
+    }
+
+    /**
+     * @param int $customerId
+     * @return mixed|null
+     * @throws \Doctrine\ORM\NoResultException
+     * @throws \Doctrine\ORM\NonUniqueResultException
+     */
+    private function getCustomerDefaultNonApplePayPaymentMethod(int $customerId) {
```

Comment:
please do not use sql or queries in a subscriber
please place it in a service / component..just anywhere else
theres an Account componentn, that would be a great thing i guess...to get xy from the current account
that also helps us to do some refactoring later on when restructuring all these services

Reply:
Done